src/resolvers/acl.js

```diff
@@ -37,5 +39,134 @@ export default {
 
       return dataSources.aclSourceDelete(args, context, parseResolveInfo(info))
     }
+  },
+
+  Acl: {
+    groups: async (source, args, context, info) => {
+      const { dataSources } = context
+      const resolvedInfo = parseResolveInfo(info)
+      const requestInfo = parseRequestedFields(resolvedInfo, {}, 'aclGroup')
+
+      const { groups } = source
+
+      const aclGroupResponse = camelcaseKeys(groups, { deep: true })
```

Comment:
We need to add a check here to prevent unnecessary calls to the EDL API. Essentially, check `requestInfo.jsonKeys` which will be an array of the keys the user requested and compare it to the data we already have available in the source. 

```
const sourceKeys = ['id', 'permissions', 'userType']

const { jsonKeys } = requestInfo

// If all the keys requested are in the source, return the source
if (jsonKeys.every((requestedKey) => sourceKeys.includes(requestedKey))) {
  return {
    count: aclGroupResponse.length,
    items: aclGroupResponse.map((group) => {
      const { groupId, ...originalGroup } = group

      if (groupId) {
        return {
          ...originalGroup,
          id: groupId
        }
      }

      return originalGroup
    })
  }
}
```